boards/arduino-nano-33-ble-sense/board.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C)  2024 Méwen Berthelot
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_arduino-nano-33-ble-sense
+ * @{
+ *
+ * @file
+ * @brief       Board specific initialization for Arduino Nano 33 BLE sense
+ *
+ * @author      Méwen Berthelot <berthelotmewen@gmail.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+
+#if defined(APDS9960) || defined(HTS221) || defined(lps22hb)
+#include "periph/gpio.h"
+#endif
+
+void board_init(void)
+{
+
+    /**
+     *   VDD needs to be set in order to have the sensors powered
+     *   It also seems that if the internal I2C pins are not set as input,
+     *   all the sensors connected will not work
+     */
```

Comment:
```suggestion
    /*
     *  VDD needs to be set in order to have the sensors powered.
     *  It also seems that if the internal I2C pins are not set as input,
     *  all the sensors connected will not work.
     */
```

Also, I would have expected the i2c driver to do this already automatically, maybe someone more knowledgeable than me in board porting can comment?

Reply:
@benpicco ?

Reply:
Indeed, this had me surprised too - the I2C driver will re-configure those pins anyway, so it's weird you are setting them to input here.

Reply:
Maybe some of the drivers that were tested don't call the i2c_init function and therefore the pins are not correctly set (because the init_pins function is not called)?
https://github.com/RIOT-OS/RIOT/blob/master/cpu/nrf5x_common/periph/i2c_nrf52_nrf9160.c#L187

For example the lsm6dsxx driver does not call i2c_init: https://github.com/RIOT-OS/RIOT/blob/master/drivers/lsm6dsxx/lsm6dsxx.c#L52-L109

On the other hand there are drivers that call i2c_init such as this one: https://github.com/RIOT-OS/RIOT/blob/master/drivers/pca9633/pca9633.c#L69

When I worked on the AT24Cxx driver, it worked with the nRF52840 on the nRF52840DK, but the driver does not call i2c_init either. Maybe the default configuration for pins in the nRF52840DK is Input and since the arduino-nano-33-ble-sense has a bootloader, they might be configured differently (not Input)?

This is just a guess though, I don't have an Arduino Nano 33 BLE Sense board to test the hypothesis.